html/pfappserver/lib/pfappserver/PacketFence/Controller/Config/Switch.pm

```diff
@@ -274,6 +275,73 @@ sub invalidate_cache :Chained('object') :PathPart('invalidate_cache') :Args(0) {
     $c->response->status(200);
 }
 
+=head1 import_csv
+
+A method to be able to import switches from a CSV
+
+=cut
+
+sub import_csv :Local :Args(0) :AdminRole('SWITCHES_CREATE') {
+    my ( $self, $c ) = @_;
+   
+    my $logger = pf::log->get_logger();
+    my $upload = $c->req->upload('importcsv');
+    my $file = $upload->fh;
+    my $model = $c->model("Config::Switch");
+    my $model_group = $c->model("Config::SwitchGroup");
+
+    my $skip = 0;
+    my $skip1 = 0;
+    my $switches = 0;
+    my %seen;
+    while (my $line = <$file>) {
+        chomp $line;
+
+        unless($skip1) {
+            $skip1 = 1;
+            next;
+        }
+
+        my @fields = split "," , $line;
+        if (@fields < 3) {
+            $skip++;
+            $logger->warn("This entry has been skipped because this line: $line contains more fields than required");
+            next;
+        }
+
+        my $hostname = $fields[0];
+        $hostname =~ s/[^a-zA-Z0-9 _-]//g;
+        $hostname =~ tr/\r\n//d;
+
+        my $switch_ip = $fields[1];
+        # Don't want to process them twice...
+        my ( $status, $msg ) = $model->hasId($switch_ip);
+        if (is_success($status)) {
+            $skip++;
+            $logger->warn("This entry has been skipped because this IP: $switch_ip is existing in the switch configuration file.");
+            next;
+        }
+    
+        my $switch_group = $fields[2];
+        ( $status, $msg ) = $model_group->hasId($switch_group);
+        if (is_error($status)) {
+            $skip++;
+            $logger->warn("This entry has been skipped because the switch group: $switch_group does not exist in the switch configutaion.");
+            next;
+        }
+
+        my $assignements = {
+            description => $hostname,
+            group => $switch_group,
+        };
+
+        $model->create($switch_ip, $assignements);
+        $switches++;
+    }
+    $model->commit();
+    $c->stash( status_msg => "$switches switches have been imported and $skip switches have benn skipped." );
```

Comment:
The status message is not translated
Suggested verbiage for message (taken from node import)
```perl
$c->stash( status_msg => $c->loc("[_1] switches imported, [_2] skipped", $switches, $skip);
```